src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts

```diff
@@ -428,14 +427,39 @@ export class FileUploadAssessmentComponent implements OnInit, AfterViewInit, OnD
 
     private checkPermissions() {
         this.isAssessor = this.result && this.result.assessor && this.result.assessor.id === this.userId;
+        if (this.exercise) {
+            this.isAtLeastInstructor = this.accountService.isAtLeastInstructorInCourse(this.exercise.course || this.exercise.exerciseGroup!.exam!.course);
+        } else {
+            this.isAtLeastInstructor = this.accountService.hasAnyAuthorityDirect(['ROLE_INSTRUCTOR', 'ROLE_ADMIN']);
```

Comment:
Why do we need the else case here? I don't think we should award everyone with 'ROLE_INSTRUCTOR' these rights in this case.